framework-crates/objc2-foundation/src/thread.rs

```diff
@@ -41,7 +41,22 @@ pub fn is_multi_threaded() -> bool {
 /// Whether the current thread is the main thread.
 #[cfg(feature = "NSThread")]
 pub fn is_main_thread() -> bool {
-    NSThread::isMainThread_class()
+    #[cfg(feature = "libc")]
+    use libc::pthread_main_np;
```

Comment:
This is not always available on GNUStep, so I suspect you might need to keep using `NSThread::isMainThread_class()` on `feature = "gnustep-1-7"` to make CI happy?